core/iwasm/common/gc/gc_object.c

```diff
@@ -311,6 +324,27 @@ wasm_externref_obj_new(WASMExecEnv *exec_env, void *heap_handle, void *host_obj)
     return externref_obj;
 }
 
+WASMAnyrefObjectRef
+wasm_anyref_obj_new(WASMExecEnv *exec_env, void *heap_handle, void *host_obj)
+{
+    WASMAnyrefObjectRef anyref_obj;
+    WASMLocalObjectRef local_ref;
+
+    if (!(anyref_obj = gc_obj_malloc(heap_handle, sizeof(WASMAnyrefObject)))) {
+        return NULL;
+    }
+
+    anyref_obj->header = WASM_OBJ_ANYREF_OBJ_FLAG;
+    anyref_obj->host_obj = host_obj;
+
+    /* Lock anyref_obj in case it is reclaimed when allocating memory below */
+    wasm_runtime_push_local_object_ref(exec_env, &local_ref);
```

Comment:
No need to push/pop local object ref here, we added in wasm_externref_obj_new since there is second `gc_obj_malloc` to create the externref_obj, which may trigger GC reclaim to free the anyref_obj allocated by the first `gc_obj_malloc`. We need to lock the first object created.